### PR TITLE
[225] Refactor init command

### PR DESCRIPTION
**Pull request by alexisbcc**

### CODE REVIEW

File: hapistrano.cabal [hunk 1]
```diff
@@ -58,6 +58,7 @@ library
                      , filepath           >= 1.2 && < 1.5
                      , gitrev             >= 1.2 && < 1.4
                      , mtl                >= 2.0 && < 3.0
+                     , megaparsec
```
Comment (CristhianMotoche) [inline hapistrano.cabal:61]: @alexisbcc Can you please specify the boundaries of this library?

File: src/System/Hapistrano.hs [hunk 2]
```diff
@@ -276,36 +278,82 @@ initConfig getLine' = do
...
+      userInput <- prompt' (parameterName <> " (default: " <> show def <> ")")
+      if null userInput then
+        pure def
+      else do
+        let parsed = M.parse (parser <* M.eof) "" userInput
+        case parsed of
+          Left err -> do
+            hPutStrLn stderr (M.errorBundlePretty  err)
+            prompt parameterName def parser
+          Right res -> pure res
```
Comment (CristhianMotoche) [inline src/System/Hapistrano.hs:303]: @alexisbcc What do you think about something like this:
```suggestion
      when (null userInput)
        pure def

      either
          (\err -> hPutStrLn stderr (M.errorBundlePretty err) >> prompt parameterName def parser)
          pure
          (M.parse (parser <* M.eof) "" userInput)
```
?

File: src/System/Hapistrano.hs [hunk 3]
```diff
@@ -276,36 +278,82 @@ initConfig getLine' = do
...
+
+    promptYN = do
+      userInput <- prompt "Include restart command? y/N" 'N' yNParser
+      case toLower userInput of
+        'y' -> pure $ Just "echo 'Restart command'"
+        _ -> pure Nothing
```
Comment (CristhianMotoche) [inline src/System/Hapistrano.hs:309]:
```suggestion
      pure $ case toLower userInput of
        'y' -> Just "echo 'Restart command'"
        _ -> Nothing
```

File: src/System/Hapistrano.hs [hunk 4]
```diff
@@ -276,36 +278,82 @@ initConfig getLine' = do
...
+
+    prompt' :: String -> IO String
+    prompt' title = do
+        hPutStrLn stderr title
+        getLine'
```
Comment (CristhianMotoche) [inline src/System/Hapistrano.hs:314]:
```suggestion
    prompt' title = hPutStrLn stderr title >> getLine'
```

Comment (CristhianMotoche) [reply src/System/Hapistrano.hs:314]: Why printing the title in `strerr`? What about using `putStrLn`?
```suggestion
    prompt' title = putStrLn title >> getLine'
```

File: src/System/Hapistrano.hs (hunk 2 above)
Comment (alexisbcc) [reply src/System/Hapistrano.hs:303]: Looks better thanks

File: src/System/Hapistrano.hs (hunk 4 above)
Comment (alexisbcc) [reply src/System/Hapistrano.hs]: I'm not sure about this one, I was reusing that part of the implementation :thinking:

**Pull request merged**